arangod/GeneralServer/AcceptorTcp.cpp

```diff
@@ -126,6 +126,7 @@ void AcceptorTcp<T>::open() {
 
 template <SocketType T>
 void AcceptorTcp<T>::close() {
+  std::unique_lock<std::mutex> guard(_mutex);
   if (_asioSocket) {
     _asioSocket->timer.cancel();
   }
```

Comment:
I don't like holding the mutex for the entire duration of `close()`, but it looks safe and like it won't block any other place that also grabs this mutex. So I guess this works.